src/analysis/retail/shaman/restoration/modules/dragonflight/Tier31.tsx

```diff
@@ -0,0 +1,151 @@
+import ITEMS from 'common/ITEMS';
+import talents from 'common/TALENTS/shaman';
+import { TIERS } from 'game/TIERS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { HealEvent } from 'parser/core/Events';
+import { isFrom4pcT31 } from '../../normalizers/CastLinkNormalizer';
+import { SHAMAN_T31_ID } from 'common/ITEMS/dragonflight';
+import { SpellLink } from 'interface';
+import ItemSetLink from 'interface/ItemSetLink';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import Statistic from 'parser/ui/Statistic';
+import { formatNumber, formatPercentage } from 'common/format';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import RiptideTracker from '../core/RiptideTracker';
+import Combatants from 'parser/shared/modules/Combatants';
+
+const RIPTIDE_4P_POWER = 0.2; // Riptide bonus power = 25% => Tier set value = 25%/125% = 20% of total amount
+//const debug = false;
+
+/**
+ * **Resto Shaman T31 (Amirdrassil)**
+ *
+ * 2pc: Chain Heal, Healing Surge, and Healing Wave mark their initial target with a Tidal Reservoir, causing them to receive 15% of all Riptide healing you deal for 15 sec.
+ *
+ * 4pc : Riptide's healing is increased by 25%. If Riptide is active on the same target as Tidal Reservoir, its heal over time effect has a 6% chance to create a new Riptide on a nearby ally.
+ */
+
+export default class Tier31 extends Analyzer {
+  static dependencies = {
+    riptideTracker: RiptideTracker,
+    combatants: Combatants,
+  };
+  protected riptideTracker!: RiptideTracker;
+  protected combatants!: Combatants;
+  has4pc: boolean;
+  tidalReservoirHealing: number = 0;
+  tidalReservoirOverhealing: number = 0;
+  bonus4pcRiptideHealing: number = 0;
+  percent4pcRiptideHealing: number = 0;
+  bonus4pcRiptideAmount: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.has2PieceByTier(TIERS.T31);
+    this.has4pc = this.selectedCombatant.has4PieceByTier(TIERS.T31);
+
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(talents.RIPTIDE_TALENT),
+      this.handleRiptideHeal,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(ITEMS.T31_TIDAL_RESERVOIR_HEAL),
+      this.on2pcHeal,
+    );
+  }
+
+  get twoPieceOverhealPercent() {
+    return formatPercentage(
+      this.tidalReservoirOverhealing /
+        (this.tidalReservoirHealing + this.tidalReservoirOverhealing),
+    );
+  }
+
+  get fourPieceTotalHealing() {
+    return this.bonus4pcRiptideHealing + this.percent4pcRiptideHealing;
+  }
+
+  on2pcHeal(event: HealEvent) {
+    this.tidalReservoirHealing += event.amount + (event.absorbed || 0);
+    this.tidalReservoirOverhealing += event.overheal || 0;
+  }
+
+  handleRiptideHeal(event: HealEvent) {
+    if (this.has4pc) {
+      const targetId = event.targetID;
```

Comment:
This method should be rewritten imo, use your attribution check to determine if the entire hot is from 4pc and if it is tally all of the healing, otherwise just tally the effective 25% increase,. something like:

`if(isFrom4pcT31(event)) {
  this.fourPieceHealing += event.amount + (event.absorbed || 0); 
}
else {
 this.fourPieceHealing += calculateEffectiveHealing(event, RIPTIDE_4P_POWER);
}`

You don't need to check for ticks if the tooltip is accurate - the bonus applies to all riptide healing 
You also don't need separate variables for the 4pc healing this way, just use one `this.fourPieceHealing`